numpy/core/src/umath/override.c

```diff
@@ -342,6 +342,29 @@ PyUFunc_CheckOverride(PyUFuncObject *ufunc, char *method,
     }
 
     /*
+     * Raise an error if an argument disables ufuncs.
+     */
+    for (i = 0; i < noa; i++) {
```

Comment:
Seems a shame not to make this happen in `PyUFunc_WithOverride`. Maybe return -1?